test/runtests.jl

```diff
@@ -406,6 +410,68 @@ function test_potential_energy_equation_terms(model; geopotential_height = nothi
         end
     end
 
+    Eb = BackgroundPotentialEnergy(model; geopotential_height)
+
+    Eb_field = Field(Eb)
+    @test Eb isa AbstractOperation
+    @test Eb_field isa Field
+    compute!(Eb_field)
+
+    return nothing
+end
+function test_1D_reference_field(model; geopotential_height = 0)
+
+    if model.buoyancy isa BuoyancyTracerModel
+
+        b = reshape(interior(model.tracers.b), :)
+        p = sortperm(b, rev = true)
+        b_sorted = b[p]
+
+        v_grid = VolumeField(model.grid)
+        A = sum(AreaField(model.grid))
+        z✶_check = -cumsum(reshape(v_grid, :)[p]) / A
+
+        sorted_buoyancy, z✶ = OneDReferenceField(model.tracers.b, rev = true)
+
+        @test isequal(reshape(interior(sorted_buoyancy), :), b_sorted)
+        @test isequal(reshape(interior(z✶), :), z✶_check)
```

Comment:
When comparing `b_sorted` and `sorted_buoiyancy`, it's always gonna return true by construction, right? Since `b_sorted` is calculated with the permutations that are used to calculate `sorted_buoyancy`? I think we should still keep this test (since it tests that the code is working and all), but maybe we need another one that's a little more rigorous.

Reply:
Would starting with linear buoyancy distribution in the `x` direction (where we know `db/dx`) and sorting it, making sure that `db/dz` matches, work? We could try that for grids that are stretched in the horizontal and vertical directions separately. If we could test something along those lines it would be great.